File: src/main/java/me/Qball/Wild/GUI/HookGui.java
package me.Qball.Wild.GUI;

import java.util.ArrayList;
import java.util.Collections;

import org.bukkit.Bukkit;
import org.bukkit.Material;
import org.bukkit.entity.Player;
import org.bukkit.inventory.Inventory;
import org.bukkit.inventory.ItemStack;
import org.bukkit.inventory.meta.ItemMeta;

public class HookGui {
    public static void openHook(Player p) {
        ItemStack close = new ItemStack(Material.REDSTONE_BLOCK, 1);
        ItemMeta meta = close.getItemMeta();
        meta.setDisplayName("Close");
        ArrayList<String> lore = new ArrayList<String>();
        lore.add("Click to close the inventory and return to normal gameplay");
        meta.setLore(lore);
        close.setItemMeta(meta);
        Inventory wildTp = Bukkit.createInventory(p, 27, "Hooks");
        p.openInventory(wildTp);
        wildTp.setItem(0, MainGui.makeItem(Material.DIAMOND_PICKAXE, "Towny Hook", Collections.singletonList("Click to enable or disable Towny Hook")));
        wildTp.setItem(2, MainGui.makeItem(Material.TNT, "Factions Hook", Collections.singletonList("Click to enable or disable Factions hook")));
        wildTp.setItem(4, MainGui.makeItem(Material.WOOD_SPADE, "GreifPrevention Hook", Collections.singletonList("Click to enable or disable Grief Prevention Hoo")));
        wildTp.setItem(6, MainGui.makeItem(Material.WOOD_AXE, "WorldGuard Hook", Collections.singletonList("Click to enable or disable WorldGuard Hook")));
        wildTp.setItem(8, MainGui.makeItem(Material.WOOD_AXE, "Kingdom Hook", Collections.singletonList("Click to enable or disable Kingdom Hook")));
        wildTp.setItem(10, MainGui.makeItem(Material.TNT, "FactionsUUD Hook", Collections.singletonList("Click to enable or disable FactionsUuid Hook")));
        wildTp.setItem(12, MainGui.makeItem(Material.GOLD_SPADE, "Residence Hook", Collections.singletonList("Click to enable or disable Residence Hook")));
        wildTp.setItem(14, MainGui.makeItem(Material.GOLD_AXE, "LandLord Hook", Collections.singletonList("Click to enable or disable LandLord Hook")));
        wildTp.setItem(16, MainGui.makeItem(Material.TNT, "LegacyFactions Hook", Collections.singletonList("Click to enable or disable LegacyFactions hook")));
        wildTp.setItem(18, MainGui.makeItem(Material.DIAMOND_AXE,"Feudal Hook", Collections.singletonList("Click to enable or disable Feudal hook")));
        wildTp.setItem(24, MainGui.backItem());
        wildTp.setItem(26, close);
    }
}

File: src/main/kotlin/me/qball/wild/gui/HookGui.kt
package me.qball.wild.gui

import org.bukkit.Bukkit
import org.bukkit.Material
import org.bukkit.entity.Player
import org.bukkit.inventory.ItemStack
import java.util.ArrayList

object HookGui {
    fun openHook(p: Player) {

        val Close = ItemStack(Material.REDSTONE_BLOCK, 1)
        val meta = Close.itemMeta
        meta.displayName = "Close"
        val lore = ArrayList<String>()
        lore.add("Click to close the inventory and return to normal gameplay")
        meta.lore = lore
        Close.itemMeta = meta
        val Wildtp = Bukkit.createInventory(p, 18, "Hooks")
        p.openInventory(Wildtp)
        Wildtp.setItem(0, towny())
        Wildtp.setItem(2, factions())
        Wildtp.setItem(4, griefPreven())
        Wildtp.setItem(6, worldGuard())
        Wildtp.setItem(8, kingdoms())
        Wildtp.setItem(17, Close)
    }

    fun factions(): ItemStack {
        val factions = ItemStack(Material.TNT, 1)
        val meta = factions.itemMeta
        meta.displayName = "Factions Hook"
        val lore = ArrayList<String>()
        lore.add("Click to enable or disable factions hook")
        meta.lore = lore
        factions.itemMeta = meta
        return factions
    }

    fun griefPreven(): ItemStack {
        val griefPreven = ItemStack(Material.WOOD_SPADE, 1)
        val meta = griefPreven.itemMeta
        meta.displayName = "GriefPrevention Hook"
        val lore = ArrayList<String>()
        lore.add("Click to enable or disable Grief Prevention Hook")
        meta.lore = lore
        griefPreven.itemMeta = meta
        return griefPreven
    }

    fun towny(): ItemStack {

        val towny = ItemStack(Material.DIAMOND_PICKAXE, 1)
        val meta = towny.itemMeta
        meta.displayName = "Towny Hook"
        val lore = ArrayList<String>()
        lore.add("Click to enable or disable Towny Hook")
        meta.lore = lore
        towny.itemMeta = meta
        return towny
    }

    fun worldGuard(): ItemStack {

        val worldGuard = ItemStack(Material.WOOD_AXE, 1)
        val meta = worldGuard.itemMeta
        meta.displayName = "WorldGuard Hook"
        val lore = ArrayList<String>()
        lore.add("Click to enable or disable WorldGuard Hook")
        meta.lore = lore
        worldGuard.itemMeta = meta
        return worldGuard
    }

    fun kingdoms(): ItemStack {
        val kingdom = ItemStack(Material.WOOD_AXE, 1)
        val meta = kingdom.itemMeta
        meta.displayName = "Kingdom Hook"
        val lore = ArrayList<String>()
        lore.add("Click to enable or disable Kingdom Hook")
        meta.lore = lore
        kingdom.itemMeta = meta
        return kingdom
    }
}

